numba/cuda/tests/cudapy/test_enums.py

```diff
@@ -0,0 +1,116 @@
+"""
+Test cases adapted from numba/tests/test_enums.py
+"""
+
+import numpy as np
+
+from numba import int8, int16, int32
+from numba import cuda, vectorize
+from numba.tests.support import TestCase
+from numba.tests.enum_usecases import (
+    Color,
+    Shape,
+    Planet,
+    RequestError,
+    IntEnumWithNegatives
+)
+
+
+class EnumTest(TestCase):
+
+    pairs = [
+        (Color.red, Color.red),
+        (Color.red, Color.green),
+        (Planet.EARTH, Planet.EARTH),
+        (Planet.VENUS, Planet.MARS),
+        (Shape.circle, IntEnumWithNegatives.two) # IntEnum, same value
+    ]
+
+    def test_compare(self):
+        @cuda.jit
+        def f(a, b, out):
+            out[0] = a == b
+            out[1] = a != b
+            out[2] = a is b
+            out[3] = a is not b
+
+        arr = np.zeros((4,), dtype=np.bool_)
+        for a, b in self.pairs:
+            f[1, 1](a, b, arr)
+            self.assertPreciseEqual(arr, np.array([
+                a == b, a != b, a is b, a is not b
+            ]))
+
+    def test_getattr_getitem(self):
+        @cuda.jit
+        def f(out):
+            # Lookup of a enum member on its class
+            out[0] = Color.red == Color.green
+            out[1] = Color['red'] == Color['green']
+
+        arr = np.zeros((2,), dtype=np.bool_)
+        f[1, 1](arr)
+        self.assertPreciseEqual(arr, np.array([
+            Color.red == Color.green, Color['red'] == Color['green']
+        ]))
+
+    def test_return_from_device_func(self):
+        @cuda.jit
+        def helper(pred):
+            return Color.red if pred else Color.green
+
+        @cuda.jit
+        def f(pred, out):
+            out[0] = helper(pred) == Color.red
+            out[1] = helper(not pred) == Color.green
+
+        arr = np.zeros((2,), dtype=np.bool_)
+        f[1, 1](True, arr)
+        self.assertPreciseEqual(arr, np.array([
+            Color.red == Color.red, Color.green == Color.green
+        ]))
+
+    def test_int_coerse(self):
```

Comment:
Possibly a typo from the original?

```suggestion
    def test_int_coerce(self):
```

Reply:
(ah, it wasn't a typo in the original :slightly_smiling_face:)